tests/puppeteer/pages/BO/orders/index.js

```diff
@@ -93,10 +113,48 @@ module.exports = class Order extends BOBasePage {
    * @return {Promise<textContent>}
    */
   async getTextColumn(columnName, row) {
+    if (columnName === 'osname') {
+      return this.getTextContent(this.updateStatusInTablebutton.replace('%ROW', row));
+    }
     return this.getTextContent(
       this.tableColumn
         .replace('%ROW', row)
         .replace('%COLUMN', columnName),
     );
   }
+
+  /**
+   * Get all row information from orders table
+   * @param row
+   * @return {Promise<object>}
+   */
+  async getOrderFromTable(row) {
+    return {
+      id: parseFloat(await this.getTextColumn('id_order', row)),
+      reference: await this.getTextColumn('reference', row),
+      newClient: await this.getTextColumn('new', row),
+      delivery: await this.getTextColumn('country_name', row),
+      customer: await this.getTextColumn('customer', row),
+      totalPaid: await this.getTextColumn('total_paid_tax_incl', row),
+      payment: await this.getTextColumn('payment', row),
+      status: await this.getTextColumn('osname', row),
+    };
+  }
+
+  /**
+   * Get order from table in csv format
+   * @param row
+   * @return {Promise<string>}
+   */
+  async getOrderInCsvFormat(row) {
+    const order = await this.getOrderFromTable(row);
+    return `${order.id};`
+      + `${order.reference};`
+      + `${order.newClient === 'Yes' ? 1 : 0};`
+      + `${order.delivery.split(' ').length > 1 ? `"${order.delivery}";` : `${order.delivery};`}`
+      + `"${order.customer}";`
+      + `${order.totalPaid};`
+      + `${order.payment.split(' ').length > 1 ? `"${order.payment}";` : `${order.payment};`}`
+      + `${order.status.split(' ').length > 1 ? `"${order.status}";` : `${order.status};`}`;
```

Comment:
I'm not sur this will be robust enough. Did you have the chance to test this with a lot of different orders to be sure it works in every case ?

Reply:
With demo data, we have only a problem with status part but I added it for delivery (United states / France) and payment method (Bank wire / Payment by check).
I can't be sure that it cover all cases.